server/issue.go

```diff
@@ -158,6 +158,159 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
 		return http.StatusInternalServerError, err
 	}
 
+	cimd, _, err := jiraClient.Issue.GetCreateMetaWithOptions(&jira.GetQueryOptions{
+		Expand: "projects.issuetypes.fields",
+	})
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to get CreateIssue metadata")
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	b, err := json.Marshal(cimd)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to marshal response")
+	}
+	_, err = w.Write(b)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to write response")
+	}
+
+	return http.StatusOK, nil
+}
+
+func httpAPIAttachIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodPost {
+		return http.StatusMethodNotAllowed,
+			errors.New("method " + r.Method + " is not allowed, must be POST")
+	}
+
+	api := ji.GetPlugin().API
+
+	comment := &struct {
+		PostId string       `json:"post_id"`
+		Fields jira.Comment `json:"fields"`
+	}{}
+	err := json.NewDecoder(r.Body).Decode(&comment)
+	if err != nil {
+		return http.StatusBadRequest,
+			errors.WithMessage(err, "failed to decode incoming request")
+	}
+
+	mattermostUserId := r.Header.Get("Mattermost-User-Id")
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	jiraUser, err := ji.GetPlugin().LoadJIRAUser(ji, mattermostUserId)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	jiraClient, err := ji.GetJIRAClient(jiraUser)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	// Lets add a permalink to the post in the Jira Description
+	post, appErr := api.GetPost(comment.PostId)
+
+	if appErr != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(appErr, "failed to load post "+comment.PostId)
+	}
+
+	mattermostUser, err := api.GetUser(post.UserId)
```

Comment:
For further understanding:

The change is only a naming convention to save the error returned from the plugin api method and to specify that the return error is typed (*model.AppError)?  

Thanks for the catch and link to the explanation.

Reply:
No, the issue is that when you wrap a non-interface nil pointer to an interface, it's not a nil value of that interface. I actually don't understand how this even worked before, but I didn't even bother since the change was worth it on stylistic grounds alone.